src/components/side-panel/layer-panel/layer-list.js

```diff
@@ -0,0 +1,179 @@
+  z-index: ${props => props.theme.dropdownWrapperZ + 1};
+
+  &.sorting {
+    pointer-events: none;
+  }
+
+  &.sorting-layers .layer-panel__header {
+    background-color: ${props => props.theme.panelBackgroundHover};
+    font-family: ${props => props.theme.fontFamily};
+    font-weight: ${props => props.theme.fontWeight};
+    font-size: ${props => props.theme.fontSize};
+    line-height: ${props => props.theme.lineHeight};
+    *,
+    *:before,
+    *:after {
+      box-sizing: border-box;
+    }
+    .layer__drag-handle {
+      opacity: 1;
+      color: ${props => props.theme.textColorHl};
+    }
+  }
+`;
+
+LayerListFactory.deps = [LayerPanelFactory];
+
+function LayerListFactory(LayerPanel) {
+  // By wrapping layer panel using a sortable element we don't have to implement the drag and drop logic into the panel itself;
+  // Developers can provide any layer panel implementation and it will still be sortable
+  const SortableItem = SortableElement(({children, isSorting}) => {
+    return (
+      <SortableStyledItem className={classnames('sortable-layer-items', {sorting: isSorting})}>
+        {children}
+      </SortableStyledItem>
+    );
+  });
+
+  const WrappedSortableContainer = SortableContainer(({children}) => {
+    return <div>{children}</div>;
+  });
+
+  const LayerList = props => {
+    const {
+      layers,
+      datasets,
+      layerOrder,
+      uiStateActions,
+      visStateActions,
+      layerClasses,
+      isSortable = true
+    } = props;
+    const {toggleModal: openModal} = uiStateActions;
+    const [isSorting, setIsSorting] = useState(false);
+
+    const layerTypeOptions = useMemo(
+      () =>
+        Object.keys(layerClasses).map(key => {
+          const layer = new layerClasses[key]();
+          return {
+            id: key,
+            label: layer.name,
+            icon: layer.layerIcon,
+            requireData: layer.requireData
+          };
+        }),
+      [layerClasses]
+    );
+
+    const layerActions = useMemo(() => {
+      return {
+        layerColorUIChange: visStateActions.layerColorUIChange,
+        layerConfigChange: visStateActions.layerConfigChange,
+        layerVisualChannelConfigChange: visStateActions.layerVisualChannelConfigChange,
+        layerTypeChange: visStateActions.layerTypeChange,
+        layerVisConfigChange: visStateActions.layerVisConfigChange,
+        layerTextLabelChange: visStateActions.layerTextLabelChange,
+        removeLayer: visStateActions.removeLayer,
+        duplicateLayer: visStateActions.duplicateLayer
+      };
+    }, [
+      visStateActions.layerColorUIChange,
+      visStateActions.layerConfigChange,
+      visStateActions.layerVisualChannelConfigChange,
+      visStateActions.layerTypeChange,
+      visStateActions.layerVisConfigChange,
+      visStateActions.layerTextLabelChange,
+      visStateActions.removeLayer,
+      visStateActions.duplicateLayer
+    ]);
+
+    const panelProps = useMemo(() => {
```

Comment:
You don't need to instantiate `panelProps` with `useMemo` `panelProps` will be spread into `<LayerPanel>`,  useMemo will do anything to improve perf

Reply:
I listened to Giuseppe so I added it. It wasn't there at first and it didn't make sense to me but ok. I will remove it.